mas-foundation/src/main/java/com/ca/mas/core/storage/sharedstorage/AccountManagerUtil.java

```diff
@@ -38,40 +39,59 @@
     public AccountManagerUtil(Context context, String accountName, boolean sharedStorage){
 
         // Gets the account type from the manifest
-        String accountType = getAccountType(context);
+        final String accountType = getAccountType(context);
+        final StringBuilder messageBuilder = new StringBuilder();
+
         if (accountType == null || accountType.isEmpty()) {
             throw new IllegalArgumentException(MASFoundationStrings.SHARED_STORAGE_NULL_ACCOUNT_TYPE);
         }
 
+        if (accountName == null) {
+            throw new IllegalArgumentException(MASFoundationStrings.SHARED_STORAGE_NULL_ACCOUNT_NAME);
+        }
+
         shared = sharedStorage;
 
         try {
-            SharedStorageIdentifier identifier = new SharedStorageIdentifier();
-
-            mAccountManager = AccountManager.get(MAS.getContext());
-            //Attempt to retrieve the account
-            Account[] accounts = mAccountManager.getAccountsByType(accountType);
-            for (Account account : accounts) {
-                if (accountName.equals(account.name)) {
-                    String password = mAccountManager.getPassword(account);
-                    String savedPassword = identifier.toString();
-                    if (password.equals(savedPassword)) {
-                        mAccount = account;
-                    }else {
-                        // - case migration from old AccountManagerStoreDataSource
-                        mAccount = null;
-                        identifier = new SharedStorageIdentifier();
+            synchronized (mutex) {
```

Comment:
usually, I use synchronized first and then try-catch block.

Rule of thump: Keep the try-catch constructs as close to the code.

Reply:
Whats wrong in synchronising entire constructor ?